sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/LockRenewSubscriber.java

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.servicebus;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.messaging.servicebus.implementation.LockContainer;
+import org.reactivestreams.Subscription;
+import reactor.core.CoreSubscriber;
+import reactor.core.publisher.BaseSubscriber;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.Operators;
+
+import java.time.Duration;
+import java.time.OffsetDateTime;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicReferenceFieldUpdater;
+import java.util.function.Function;
+
+/**
+ * Receives messages from to upstream, pushes them downstream and start lock renewal.
+ */
+final class LockRenewSubscriber extends BaseSubscriber<ServiceBusReceivedMessage> {
+    private static final OffsetDateTime COMPLETED_RENEW_TIME = OffsetDateTime.now();
+    private static final LockRenewalOperation COMPLETED_ONE = new LockRenewalOperation("lock-no-renew",
+        Duration.ZERO, false, (id) -> Mono.empty(), COMPLETED_RENEW_TIME);
+
+    private final ClientLogger logger = new ClientLogger(LockRenewSubscriber.class);
+
+    private final Function<String, Mono<OffsetDateTime>> onRenewLock;
+    private final Duration maxAutoLockRenewal;
+    private final LockContainer<LockRenewalOperation> messageLockContainer;
+    private final CoreSubscriber<? super ServiceBusReceivedMessage> actual;
+
+    private volatile Subscription subscription;
+    private static final AtomicReferenceFieldUpdater<LockRenewSubscriber, Subscription> UPSTREAM =
```

Comment:
Why are you keeping a reference to this? BaseSubscriber holds most of this logic. I don't see a call to the base class either.